src/libs/auth.js

```diff
@@ -37,6 +37,10 @@ export const hasBillingScope = () => {
   return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing')
 }
 
+const registeredAndAcceptedTos = (oldState, state) => {
+  return (oldState.registrationStatus !== 'registered' && state.registrationStatus === 'registered' && state.acceptedTos) ||
+    (state.registrationStatus === 'registered' && state.acceptedTos && !oldState.acceptedTos)
```

Comment:
I have a difficult time reasoning about the logic above. I'm sure I'm lacking some domain knowledge around registration status and ToS acceptance. Why do we care about `oldState`? That is, why can't we simply check if `state.registrationStatus === 'registered' && state.acceptedTos`?

Reply:
FWIW: Assuming we need all of the logic, the following reads a bit easier for me: 

```suggestion
  return state.registrationStatus === 'registered' && state.acceptedTos && (oldState.registrationStatus !== 'registered' || !oldState.acceptedTos)
```

Reply:
My understanding of this code (which I am still learning about) is that we only want to make these Ajax calls once per session, either at log in for someone already registered/accepted TOS, and then at the time that a new user transitions from not being registered/not having accepted TOS to "fully registered". I think that is what the "oldState/newState" checks are about. But I am going to dig into this more.